.github/workflows/actionlint.yml

```diff
@@ -0,0 +1,81 @@
+# This file is synced from the `.github` repository, do not modify it directly.
+name: Workflow Syntax
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - '.github/workflows/*.ya?ml'
+  pull_request:
+    paths:
+      - '.github/workflows/*.ya?ml'
+
+defaults:
+  run:
+    shell: bash -xeuo pipefail {0}
+
+concurrency:
+  group: "actionlint-${{ github.ref }}"
+  cancel-in-progress: ${{ github.event_name == 'pull_request' }}
+
+env:
+  HOMEBREW_DEVELOPER: 1
+  HOMEBREW_NO_AUTO_UPDATE: 1
+  HOMEBREW_NO_ENV_HINTS: 1
+
+permissions: {}
+
+jobs:
+  workflow_syntax:
+    if: github.repository_owner == 'Homebrew'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Homebrew
+        id: setup-homebrew
+        uses: Homebrew/actions/setup-homebrew@master
+        with:
+          core: false
+          cask: false
+          test-bot: false
+
+      - name: Install tools
+        run: brew install actionlint shellcheck zizmor
+
+      - uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+
+      - run: zizmor --format sarif . >results.sarif
+
+      - name: Upload SARIF file
+        uses: actions/upload-artifact@v4
+        with:
+          name: results.sarif
+          path: results.sarif
+
+      - name: Set up actionlint
+        run: echo "::add-matcher::$(brew --repository)/.github/actionlint-matcher.json"
+
+      - run: actionlint
```

Comment:
Should [ignore the matrix.container error](https://github.com/Homebrew/brew/pull/17482#issuecomment-2163812497) to avoid failures in [homebrew-portable-ruby](https://github.com/Homebrew/homebrew-portable-ruby/actions/runs/11610658511/job/32330390093?pr=248) and [actions](https://github.com/Homebrew/actions/actions/runs/11610665580/job/32330413964?pr=605).

Reply:
Pushed fixes to avoid the error instead of ignoring it -- adding `-ignore` isn't great since it means you can't run `actionlint` directly from the command-line.